resources/splunk_install.rb

```diff
@@ -3,26 +3,22 @@
 #
 # Resource for managing the installation of Splunk
 class SplunkInstall < ChefCompat::Resource
-  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers
+  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers, CernerSplunk::ResourceHelpers
 
   resource_name :splunk_install
 
   property :name, String, name_property: true, identity: true
   property :package, [:splunk, :universal_forwarder], required: true
   property :version, String, required: true
   property :build, String, required: true
+  property :user, String, default: lazy { node['current_user'] || package == :splunk ? 'splunk' : 'splunkforwarder' }
```

Comment:
I think this should just default to 'splunk' 

node['current_user'] will always be 'root' on linux as it's the effective UID of the chef-client process: https://github.com/chef/ohai/commit/fec9467a0c93b203e97447207352745536c2b471